pkg/adapter/mtping/pingsource.go

```diff
@@ -50,9 +47,17 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, source *v1beta2.PingSour
 	return nil
 }
 
-func (r *Reconciler) FinalizeKind(ctx context.Context, source *v1beta2.PingSource) reconciler.Event {
-	// Update the adapter state
-	r.mtadapter.Remove(ctx, source)
-
-	return nil
+func (r *Reconciler) deleteFunc(obj interface{}) {
+	if obj == nil {
+		return
+	}
+	acc, err := kmeta.DeletionHandlingAccessor(obj)
+	if err != nil {
+		return
+	}
+	pingSource, ok := acc.(*v1beta2.PingSource)
+	if !ok || pingSource == nil {
+		return
+	}
+	r.mtadapter.Remove(context.Background(), pingSource)
```

Comment:
While you are at it, can you change the signature of `Remove` to just pass pingSource (for consistency sake)?

Reply:
@lionelvillard Thanks for your remind, Done!